pkg/api/validate/openshiftcluster_validatedynamic.go

```diff
@@ -287,3 +294,20 @@ func (dv *openShiftClusterDynamicValidator) validateVnet(ctx context.Context, vn
 
 	return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidLinkedVNet, "", "The provided vnet '%s' is invalid: custom DNS servers are not supported.", vnetID)
 }
+
+func (dv *openShiftClusterDynamicValidator) validateProviders(ctx context.Context, providerClient resources.ProvidersClient) error {
+	providers := []string{
+		"Microsoft.Storage",
```

Comment:
we should probably also check Microsoft.Authorization, Microsoft.Compute, Microsoft.Network.